### ISSUE TITLE
bug: cannot pass interval with more than one unit to `ibis.literal`

### ISSUE THREAD

**Issue by chloeh13q** [ibis-project/ibis#6458]
### What happened?

`ibis.interval()` is not able to infer type for `IntervalAdd` objects, therefore if you pass interval with more than one unit to `ibis.literal()` it will error out.

<img width="744" alt="Screenshot 2023-06-15 at 1 52 51 PM" src="https://github.com/ibis-project/ibis/assets/57872406/229400cb-f43b-44b0-bb6d-3d371bef7f8e">

### What version of ibis are you using?

5.1.0

### Relevant log output

```sh
---------------------------------------------------------------------------
IbisTypeError                             Traceback (most recent call last)
Cell In[2], line 1
----> 1 print(ibis.literal(ibis.interval(seconds=50, minutes=1)))
      2 print(type(ibis.literal(ibis.interval(seconds=50, minutes=1))))

File ~/Desktop/ibis-public/ibis/expr/types/generic.py:1631, in literal(value, type)
   1628 if isinstance(value, Expr):
   1629     value = value.op()
-> 1631 return rlz.literal(type, value).to_expr()

File ~/Desktop/ibis-public/ibis/expr/rules.py:98, in rule.__call__(self, *args, **kwargs)
     96 args = map(self._erase_expr, args)
     97 kwargs = {k: self._erase_expr(v) for k, v in kwargs.items()}
---> 98 result = super().__call__(*args, **kwargs)
     99 assert not isinstance(result, ir.Expr)
    100 return result

File ~/anaconda3/envs/ibis/lib/python3.10/site-packages/toolz/functoolz.py:304, in curry.__call__(self, *args, **kwargs)
    302 def __call__(self, *args, **kwargs):
    303     try:
--> 304         return self._partial(*args, **kwargs)
    305     except TypeError as exc:
    306         if self._should_curry(args, kwargs, exc):

File ~/Desktop/ibis-public/ibis/expr/rules.py:189, in literal(dtype, value, **kwargs)
    187     dtype = inferred_dtype
    188 else:
--> 189     raise com.IbisTypeError(
    190         'The datatype of value {!r} cannot be inferred, try '
    191         'passing it explicitly with the `type` keyword.'.format(value)
    192     )
    194 if dtype.is_null():
    195     return ops.NullLiteral()

IbisTypeError: The datatype of value  cannot be inferred, try passing it explicitly with the `type` keyword.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jcrist** [comment]
This has already been resolved in the dev branch by #6138.

```python
In [1]: import ibis

In [2]: ibis.interval(seconds=50, minutes=1)
Out[2]: IntervalAdd(50, 1): 50 s + 1 m
```

**Comment by chloeh13q** [comment]
@jcrist I think this is a slightly different issue? `ibis.interval(seconds=50, minutes=1)` does work, but `ibis.literal(ibis.interval(seconds=50, minutes=1))` does not
